src/rgw/rgw_op.cc

```diff
@@ -2392,10 +2392,28 @@ void RGWCreateBucket::execute()
   if (op_ret < 0)
     return;
 
-  if (!store->get_zonegroup().is_master &&
+  if (!location_constraint.empty()) {
+    std::string current_period_id = store->get_current_period_id();
+    if (!current_period_id.empty()) {
+      RGWPeriod current_period(current_period_id);
+      auto zonegroups_api = current_period.get_map().zonegroups_by_api;
```

Comment:
for this to work, the period object would have to be read from rados with `RGWPeriod::init()`. but this period is already stored in `RGWRados::current_period`, so we should use that instead. it's private, so we'd need to add a helper function like `get_zonegroup_by_api()` (similar to `RGWRados::find_zone_id_by_name()`)